test/src/e2e_vm_tests/test_programs/should_pass/stdlib/alloc/src/main.sw

```diff
@@ -19,20 +18,25 @@ fn sw(ptr: raw_ptr, val: u64) {
     };
 }
 
-fn main() -> bool {
-    let hp_start = heap_ptr();
+fn heap_ptr() -> raw_ptr {
+    asm(ptr) {
+        addi ptr hp i1;
+        ptr: raw_ptr
+    }
+}
```

Comment:
What is the reason for not using `heap_ptr()` from the standard library directly?